### ISSUE TITLE
[Help] How to change NSPanel Nextion display baud rate back to 115200?

### ISSUE THREAD

**Issue by tzahari** [joBr99/nspanel-lovelace-ui#564]
### PROBLEM DESCRIPTION

I uploaded a wrong TFT file into the Nextion display. Afterwards the display showed "System data ERROR" and was not able to upload a correct TFT file. After a few checks I detected that the display runs now with 9600 baud.
Uploading of the lui-release.tft will takes forever with 9600 baud and stopped after a few percent..

How can I change the display back to 115200 baud?

I searched the internet but couldn't find a working solution.

Thanks for you help

I also created an issue at sairon/esphome-nspanel-lovelace-ui#16 but I think here is the better place?

[X] I've double checked each step explained in the FAQ:

https://docs.nspanel.pky.eu/faq/#waiting-for-content-this-is-taking-longer-than-usual-on-the-screen

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Provide a screenshot of your tasmota mqtt configuration:
not running tasmota

- [ ] Provide your appdaemon.yaml: 
```
  appdaemon.yaml here:


```

- [ ] Provide your apps.yaml:
```
  apps.yaml here:


```
- [ ] Go to Settings>Add-ons>AppDaemon>Log and then, provide the output of the log after restarting the appdaemon container:
```
  Log output here:


```

### ADDITIONAL INFORMATION

I think the above information are not required cause it is a basic issue with the Nextion display...

**Comment by joBr99** [comment]
Whats makes you think that your panel runs with 9600 baud?
The serial speed of the flashing is set during the flashing process and not configured on the panel.

The is a small test firmware that you can flash: http://nspanel.pky.eu/test.lui

If you are using ESPHome the esphome repo was in general the correct one, but let me try to understand what your problem is.

**Comment by tzahari** [comment]
Hi, thanks for your response.
Your link is not working - 404.
I use ESPhome with the example yaml. I can provide if you need it.
If in ESPhome the UART is configured to 115200 baud then this happened: 

```
[18:03:45][D][nspanel_lovelace:098]: Sending: DRAKJHSUYDGBNCJHGJKSHBDN
[18:03:45][D][nspanel_lovelace:098]: Sending: recmod=0
[18:03:45][D][nspanel_lovelace:098]: Sending: recmod=0
[18:03:45][D][nspanel_lovelace:098]: Sending: connect
[18:03:45][D][nspanel_lovelace_upload:150]: Connected
[18:03:45][D][nspanel_lovelace_upload:156]: Requesting URL: http://192.168.x.x:8123/local/lui-release.tft
[18:03:45][D][nspanel_lovelace_upload:188]: Updating Nextion
[18:03:45][D][nspanel_lovelace:098]: Sending: whmi-wris 7866204,115200,1
[18:03:45][D][nspanel_lovelace_upload:206]: Waiting for upgrade response
[18:03:47][D][nspanel_lovelace_upload:211]: Upgrade response is []
[18:03:47][D][nspanel_lovelace_upload:216]: preparation for tft update failed 0 ""
[18:03:47][D][nspanel_lovelace_upload:274]: Restarting Nextion
[18:03:47][D][nspanel_lovelace:098]: Sending: rest
[18:03:49][D][nspanel_lovelace_upload:277]: Restarting esphome
```

Then I change the baud rate of the display to 9600 in ESPhome the upload begins:
```
[18:08:09][D][nspanel_lovelace:098]: Sending: DRAKJHSUYDGBNCJHGJKSHBDN
[18:08:09][D][nspanel_lovelace:098]: Sending: recmod=0
[18:08:09][D][nspanel_lovelace:098]: Sending: recmod=0
[18:08:09][D][nspanel_lovelace:098]: Sending: connect
[18:08:09][D][nspanel_lovelace_upload:150]: Connected
[18:08:09][D][nspanel_lovelace_upload:156]: Requesting URL: http://192.168.x.x:8123/local/lui-release.tft
[18:08:09][D][nspanel_lovelace_upload:188]: Updating Nextion
[18:08:09][D][nspanel_lovelace:098]: Sending: whmi-wris 7866204,9600,1
[18:08:09][D][nspanel_lovelace_upload:206]: Waiting for upgrade response
[18:08:09][D][nspanel_lovelace_upload:211]: Upgrade response is [05]
[18:08:09][D][nspanel_lovelace_upload:214]: preparation for tft update done
[18:08:09][D][nspanel_lovelace_upload:238]: Allocating buffer size 65536, Heap size is 220620
[18:08:09][D][nspanel_lovelace_upload:255]: Updating tft from "http://192.168.x.x:8123/local/lui-release.tft" with a file size of 7866204 using 65536 chunksize, Heap Size 159032
[18:08:09][D][nspanel_lovelace_upload:032]: Requesting range: bytes=0-16383
[18:08:09][D][nspanel_lovelace_upload:081]: fetched 16384 bytes
[18:08:14][D][nspanel_lovelace_upload:090]: Uploaded 0.1 %, remaining 7862108 B
[18:08:19][D][nspanel_lovelace_upload:100]: recv_string [08.00.00.00.00 (5)]
[18:08:23][D][nspanel_lovelace_upload:090]: Uploaded 0.1 %, remaining 7858012 B
INFO nspanel-wz.local: Ping timed out!
INFO Disconnected from ESPHome API for nspanel-wz.local
WARNING Disconnected from API
WARNING Can't connect to ESPHome API for nspanel-wz.local: Error resolving IP address: [Errno -2] Name or service not known
INFO Trying to reconnect to nspanel-wz.local in the background
```
The connection via ESPhome is now is gone now but the screen still updates. 
The last time I tried this, the update progress stalled on the display after some time.

I will write another comment with the result of the current upload progress...

**Comment by joBr99** [comment]
yes, the link was wrong fixed it

I've also added a button on the page to set default baud of the panel to 115200

This is the correct link: http://nspanel.pky.eu/test.tft

Flashing of the release tft should also set baud to 115200, but it will take forever with 9600 baud.

**Comment by tzahari** [comment]
Thanks for the link. 
The upload of lui-release.tft was successful. But took a while.
The previous attempt did not work.

It seems that the display is now back on 115200 baud.

Maybe add a recovery entry to the FAQ or so?

Thanks again for your help.

**Comment by joBr99** [comment]
This is a esphome thing with my berry driver it would be possible to edit baud in the driver without affecting the baud during flashing. Looks like the esphome component is using the same baud for communication and flashing.

**Comment by mattzeer** [comment]
Hello,

i guess i have the same problem 
With baudrate at 9600, the screen change try to upload but stuck during the upload

```
[18:05:55][D][nspanel_lovelace:098]: Sending: DRAKJHSUYDGBNCJHGJKSHBDN
[18:05:55][D][nspanel_lovelace:098]: Sending: recmod=0
[18:05:55][D][nspanel_lovelace:098]: Sending: recmod=0
[18:05:55][D][nspanel_lovelace:098]: Sending: connect
[18:05:55][D][nspanel_lovelace_upload:150]: Connected
[18:05:55][D][nspanel_lovelace_upload:156]: Requesting URL: http://nspanel.pky.eu/test.tft
[18:05:55][D][nspanel_lovelace_upload:188]: Updating Nextion
[18:05:55][D][nspanel_lovelace:098]: Sending: whmi-wris 7925436,9600,1
[18:05:55][D][nspanel_lovelace_upload:206]: Waiting for upgrade response
[18:05:55][D][nspanel_lovelace_upload:211]: Upgrade response is [05]
[18:05:55][D][nspanel_lovelace_upload:214]: preparation for tft update done
[18:05:55][D][nspanel_lovelace_upload:238]: Allocating buffer size 8192, Heap size is 43456
[18:05:55][D][nspanel_lovelace_upload:243]: Allocating 4096 buffer
[18:05:55][D][nspanel_lovelace_upload:255]: Updating tft from "http://nspanel.pky.eu/test.tft" with a file size of 7925436 using 4096 chunksize, Heap Size 46668
[18:05:55][D][nspanel_lovelace_upload:032]: Requesting range: bytes=0-4095
[18:05:55][D][nspanel_lovelace_upload:081]: fetched 4096 bytes
[18:05:59][D][nspanel_lovelace_upload:090]: Uploaded 0.1 %, remaining 7921340 B
[18:06:05][D][nspanel_lovelace_upload:100]: recv_string [08.00.00.00.00 (5)]
[18:06:05][D][nspanel_lovelace_upload:266]: Heap Size 43644, Bytes left 7921340
[18:06:06][D][nspanel_lovelace_upload:032]: Requesting range: bytes=4096-8191
[18:06:06][D][nspanel_lovelace_upload:081]: fetched 4096 bytes
INFO entree-nspanel.local: Ping timed out!
INFO Disconnected from ESPHome API for entree-nspanel.local
WARNING Disconnected from API
WARNING Can't connect to ESPHome API for entree-nspanel.local: Hello timed out
INFO Trying to reconnect to entree-nspanel.local in the background
```

@tzahari how did you manage the entire upload ?

**Comment by tzahari** [comment]
Hi @mattzeer, 
it seems like my problem.
Between my tries I did several things:
I configured the tty of the display to 9600 baud and installed the firmware via the ESPhome.
I also downloaded the lui-release.tft file and put it on a webserver in my LAN.
Afterwards the upload of the file was successful but took several hours. 
I checked a few times if the display still shows up counting numbers. 
Hope that helps.
